puppet-client/nd_backup/06b_puppet_config.sh

```diff
@@ -0,0 +1,215 @@
+#       V5+ will just hit our main endpoint.
+#
+#   PUPPET_CHALLENGE_PASSWORD:
+#     Puppet Challenge Passphrase
+#
+#   PUPPET_ENVIRONMENT:
+#       The Puppet environment to use. Note, this is only used on Puppet V3
+#       installations.
+#
+#   PUPPET_ROLE:
+#       The puppet node-name to use to configure the host on bootup.
+#
+#   PUPPET_SUBTYPE:
+#       The subtype used to more narrowly define the parameters that are
+#       associated with the $PUPPET_ROLE above.
+#
+#   PUPPET_FACTS:
+#       A comma-separated list of KEY=VALUE pairs that will be set as Puppet
+#       Facts for the host. These are used to define the host type, setup
+#       custom host behaviors and more.
+#
+#   PUPPET_TRUSTED_FACTS:
+#       A comma-separated list of KEY=VALUE pairs that will be turned into
+#       attributes added to the /etc/puppet/csr_attributes.yaml file. These
+#       attributes are used by Project Zuul to deploy credentials to hosts
+#       securely.
+#
+#   INSTANCE_ID:
+#     The cloud-specific Instance ID for the host.
+#
+#   IMAGE_NAME:
+#     The cloud-specific image ID
+
+#   DATACENTER:
+#     The cloud-specific Datacenter/Region for this host.
+#
+#   ZONE:
+#      The cloud-specific zone for this host
+#
+
+# http://urbanautomaton.com/blog/2014/09/09/redirecting-bash-script-output-to-syslog/
+exec 1> >(logger -s -t puppet_config) 2>&1
+
+set -e
+
+# Figure out what apt package we need to install to get the right repo
+. /etc/profile
+PUPPET_VERSION=$(puppet --version)
+case ${PUPPET_VERSION:0:1} in
+  3) PUPPET_DIR=/etc/puppet
+     ;;
+  5) PUPPET_DIR=/etc/puppetlabs/puppet
+     ;;
+  *) echo 'Invalid Puppet Version Supplied' && exit 1 ;;
+esac
+
+FACT_DIR=/etc/facter/facts.d
+FACT_FILE=$FACT_DIR/example.txt
+CSR_ATTRS=${PUPPET_DIR}/csr_attributes.yaml
+PUPPET_ROLE=${PUPPET_ROLE:-None}
+PUPPET_SUBTYPE=${PUPPET_SUBTYPE:-}
+PUPPET_SERVER=${PUPPET_SERVER:-puppet.service.example.com}
+PUPPET_ENVIRONMENT=${PUPPET_ENVIRONMENT:-production}
+PUPPET_CA_SERVER=${PUPPET_CA_SERVER:-puppetca.example.com}
+
+# Some day this will allow for outside provisioners - and this variable will be
+# used for a host to tell us what kind of provisioner launched it, so that we
+# can use different verification methods.
+PROVISIONER=${PROVISIONER:-rightscale}
+
+PRIMARY_NIC=$(route | grep default | awk '{print $8}') # eth0 or ens3
+MAC=${MAC:-$(cat /sys/class/net/${PRIMARY_NIC}/address)}
+
+# These facts can be overridden in RightScale - but we default to using the
+# Amazon Metadata service to get the info. This will allow us in the future to
+# pass in options to this script for different clouds.. but its not necessary
+# for now.
+IMAGE_NAME=${IMAGE_NAME:-$(curl -s http://169.254.169.254/latest/meta-data/ami-id)}
+INSTANCE_ID=${INSTANCE_ID:-$(curl -s http://169.254.169.254/latest/meta-data/instance-id)}
+ZONE=${ZONE:-$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)}
```

Comment:
what happens when this runs on non-aws clouds?

Reply:
All of the stuff in nd_backup is just for reference. I am unsure how people will want to use it, and I don't want to lose the code until we are sure it works in their use-cases. The only thing running is the stuff defined in the server_template.yml